Tasks/AzureResourceManagerTemplateDeploymentV3/operations/Utils.ts

```diff
@@ -407,6 +422,62 @@ class Utils {
     private static escapeBlockCharacters(str: string): string {
         return str.replace(/[\[]/g, '$&[]');
     }
+
+    private static async getFilePathForLinkedArtifact(filePath: string): Promise<string> {
+        var filePathExtension: string = filePath.split('.').pop();
+        if(filePathExtension === 'bicep'){
+            let azcliversion = await this.getAzureCliVersion()
+            if(parseFloat(azcliversion)){
+                if(this.isBicepAvailable(azcliversion)){
+                    await this.execBicepBuild(filePath)
+                    filePath = filePath.replace('.bicep', '.json')
+                    this.cleanupFileList.push(filePath)
+                }else{
+                    throw new Error(tl.loc("IncompatibleAzureCLIVersion"));
+                }
+            }else{
+                throw new Error(tl.loc("AzureCLINotFound"));
+            }
+        }
+        return filePath
+    }
+
+    private static async getAzureCliVersion(): Promise<string> {
+        let azcliversion: string = "" ;
+        try {
+            const { stdout, stderr } = await cpExec('az version');
+            if (!stderr) {
+                azcliversion = JSON.parse(stdout)["azure-cli"]
+            } else {
+                throw stderr
+            }
+        } catch (err) {
+            throw new Error(tl.loc("FailedToFetchAzureCLIVersion", err));
+        }
+        return azcliversion
+    }
+
+    private static async execBicepBuild(filePath): Promise<void> {
+        try {
+            const { stdout, stderr } = await cpExec(`az bicep build --file ${filePath}`);
```

Comment:
You need to check the exit code !== 0, rather than just whether or not something was logged to stderr, to determine if the build failed. For example, if Bicep logs a warning message, it'll be logged to stderr, but this doesn't mean the build failed.

I'd recommend adding a test case to ensure warnings don't fail the build. You could repro this easily with a test file containing an unused variable as it'll trigger a warning:
```bicep
var unusedVar = 'hello'
```

Reply:
Resolved.